pkgs/development/tools/refurb/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, fetchFromGitHub
+, python3
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "refurb";
+  version = "1.0.0";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "dosisod";
+    repo = "refurb";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-AJM/2XfaTELR3FSWJziJjSqADAkZ45NkHOZ42XqGNbw=";
+  };
+
+  nativeBuildInputs = with python3.pkgs; [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    mypy
+    mypy-extensions
+    tomli
+    typing-extensions
+  ];
+
+  checkInputs = with python3.pkgs; [
+    pytest
+    attrs
+    black
+    click
+    colorama
+    coverage
+    flake8
+    iniconfig
+    isort
+    mccabe
+    packaging
+    pathspec
+    platformdirs
+    pluggy
+    py
+    pycodestyle
+    pyflakes
+    pyparsing
+    pytest
+    pytest-cov
+    pytestCheckHook
+  ];
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace 'mypy = "^0.981"' 'mypy = "*"' \
+  '';
+
+  pythonImportsCheck = [
+    "refurb"
+  ];
+
+  meta = with lib; {
+    description = "A tool for refurbishing and modernizing Python codebases";
+    homepage = "https://github.com/dosisod/refurb";
+    license = with licenses; [ gpl3 ];
```

Comment:
https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206

Reply:
Thanks, I was not aware of the new naming. Fixed.